jwst/charge_migration/charge_migration_step.py

```diff
@@ -28,19 +29,21 @@ class ChargeMigrationStep(Step):
     def process(self, input):
 
         # Open the input data model
-        with datamodels.RampModel(input) as input_model:
-            if (input_model.data.shape[1] < 3):  # skip step if only 1 or 2 groups/integration
-                log.info('Too few groups per integration; skipping charge_migration')
-                
-                result = input_model
-                result.meta.cal_step.charge_migration = 'SKIPPED'
+        input_model = use_datamodel(input)
 
-                return result
+        if (input_model.data.shape[1] < 3):  # skip step if only 1 or 2 groups/integration
+            log.info('Too few groups per integration; skipping charge_migration')
+            
+            result = input_model
+            result.meta.cal_step.charge_migration = 'SKIPPED'
 
-            # Retrieve the parameter value(s)
-            signal_threshold = self.signal_threshold
+            return result
 
-            result = charge_migration.charge_migration(input_model, signal_threshold)
-            result.meta.cal_step.charge_migration = 'COMPLETE'
+        # Retrieve the parameter value(s)
+        signal_threshold = self.signal_threshold
 
+        result = charge_migration.charge_migration(input_model, signal_threshold)
+        result.meta.cal_step.charge_migration = 'COMPLETE'
+
+        input_model.close()
```

Comment:
When `input` is a model, this will close the input file. This can cause issues for certain files (if the input file is asdf and lazy loaded).

Reply:
I think this is addressed by using the "with" suggested by Melanie - already in the works

Reply:
Thanks! I see that some of the steps are updated to use `with`. When the input is not cloned the `__exit__` of the with statement will close the file (if the model is the only one that has [file_references](https://github.com/spacetelescope/stdatamodels/blob/18d07d732892ddcdbb8f13098468491edac3d473/src/stdatamodels/model_base.py#L180) to the input file). These references are used to track when the file can be safely closed. When a model is cloned it [increments the reference](https://github.com/spacetelescope/stdatamodels/blob/18d07d732892ddcdbb8f13098468491edac3d473/src/stdatamodels/model_base.py#L414) to the file so that both the clone and original model need to be freed before the file is closed. This cloning and file reference counting  is what allowed these steps to use a `with` context without closing the file too early. If we want to remove the cloning and keep the file references up-to-date we'll either have to remove the with context usage or provide a way for a datamodel to be used in a with statement without requiring a clone. One option here might be to use a [nullcontext](https://docs.python.org/3/library/contextlib.html#contextlib.nullcontext) when the model is not cloned.